gui/modlistman.lua

```diff
@@ -0,0 +1,446 @@
+            earliest_compat_numeric_version = viewscreen.object_load_order_earliest_compat_numeric_version,
+            src_dir = viewscreen.object_load_order_src_dir,
+            name = viewscreen.object_load_order_name,
+            displayed_version = viewscreen.object_load_order_displayed_version,
+            mod_header = viewscreen.object_load_order_mod_header,
+        }
+    else
+        error("Invalid kind: " .. kind)
+    end
+end
+
+local function move_mod_entry(viewscreen, to, from, mod_id, mod_version)
+    local to_fields = get_modlist_fields(to, viewscreen)
+    local from_fields = get_modlist_fields(from, viewscreen)
+
+    local mod_index = nil
+    for i, v in ipairs(from_fields.id) do
+        local version = from_fields.numeric_version[i]
+        if v.value == mod_id and version == mod_version then
+            mod_index = i
+            break
+        end
+    end
+
+    if mod_index == nil then
+        return false
+    end
+
+    for k, v in pairs(to_fields) do
+        if type(from_fields[k][mod_index]) == "userdata" then
+            v:insert('#', from_fields[k][mod_index]:new())
+        else
+            v:insert('#', from_fields[k][mod_index])
+        end
+    end
+
+    for k, v in pairs(from_fields) do
+        v:erase(mod_index)
+    end
+
+    return true
+end
+
+local function enable_mod(viewscreen, mod_id, mod_version)
+    return move_mod_entry(viewscreen, "object_load_order", "available", mod_id, mod_version)
+end
+
+local function disable_mod(viewscreen, mod_id, mod_version)
+    return move_mod_entry(viewscreen, "available", "object_load_order", mod_id, mod_version)
+end
+
+local function get_active_modlist(viewscreen)
+    local t = {}
+    local fields = get_modlist_fields("object_load_order", viewscreen)
+    for i, v in ipairs(fields.id) do
+        local version = fields.numeric_version[i]
+        table.insert(t, { version = version, id = v.value })
+    end
+    return t
+end
+
+local function swap_modlist(viewscreen, modlist)
+    local current = get_active_modlist(viewscreen)
+    for _, v in ipairs(current) do
+        disable_mod(viewscreen, v.id, v.version)
+    end
+
+    local failures = {}
+    for _, v in ipairs(modlist) do
+        if not enable_mod(viewscreen, v.id, v.version) then
+            table.insert(failures, v.id)
+        end
+    end
+    return failures
+end
+
+local function is_modlist_visible()
+    -- local viewscreen = dfhack.gui.getCurViewscreen()
+    -- if viewscreen._type == df.viewscreen_new_regionst then
+    --     if viewscreen.doing_mods then
+    --         return true
+    --     end
+    -- end
+    -- return false
+
+    local vs = get_newregion_viewscreen()
+    if vs and vs.doing_mods then
+        return true
+    end
+end
+
+ModmanageMenu = defclass(ModmanageMenu, widgets.Window)
+ModmanageMenu.ATTRS {
+    view_id = "modman_menu",
+    frame_title = "Modlist Manager",
+    frame_style = gui.WINDOW_FRAME,
+    frame_inset=0,
```

Comment:
it is more common to leave this at the default of `1` and use child widget frames that have `0` offsets. In general, `frame_inset` is only set to 0 if you're doing something fancy with interior frames.